spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java

```diff
@@ -123,8 +133,17 @@ public TaskDefinitionResource save(@RequestParam("name") String name, @RequestPa
 	 */
 	@RequestMapping(value = "/{name}", method = RequestMethod.DELETE)
 	@ResponseStatus(HttpStatus.OK)
-	public void destroyTask(@PathVariable("name") String name) {
-		taskDeleteService.deleteTaskDefinition(name);
+	public void destroyTask(@PathVariable("name") String name, @RequestParam(required = false) Boolean cleanup) {
+		if (cleanup != null && cleanup.equals(Boolean.TRUE)) {
```

Comment:
Feel free to argue with me.  Shouldn't we move the consideration of deleting a task executions to the service vs. it being in the controller.

Reply:
That's a good point. Let me see if we can fit that in.